### DISCUSSION TITLE
Finding the breakpoint in Visual Studio Code

### DISCUSSION THREAD

**Discussion by imoldfella** [go-delve/delve#2912] [Ideas]
When I hit a breakpoint, its often a chore to find the right call stack in the debugger list of goroutines. Ideally the breakpoint goroutine and stack would be selected by default.

I don't know if this a Delve thing, a Gopls thing, a Visual Studio Code thing, or a I don't know what I'm doing thing. Thoughts?

**Comment by aarzilli** [comment]
cc @polinasok @suzmue but I think this is probably a vscode thing

**Comment by suzmue** [comment]
The goroutine with the breakpoint should be automatically selected by VSCode, but I have found that it can still be difficult to find in long lists of goroutines. There are a couple of ways that you can more easily find the goroutine with the breakpoint.

The selected goroutine is marked with an asterisk, to make it visually different and it will have `Paused on Breakpoint` listed as the stopped reason as pictured below.

<img width="791" alt="Screen Shot 2022-02-23 at 4 21 53 PM" src="https://user-images.githubusercontent.com/6634754/155426222-85f4073e-11c2-4774-bbba-b58a027f0df5.png">

Another suggestion I have to make it easier to find the goroutines is to use the `hideSystemGoroutines` and `goroutineFilters`. You can set these in your launch configuration to filter which goroutines appear in the call stack view and these can also be configured dynamically from the debug console.

**Comment by polinasok** [comment]
To expand on what @suzmue noted: the goroutines are returned in a specific order. The current goroutine is expanded in-place when paused on a breakpoint, so you need to scroll down the list to find it. And if this is not the first breakpoint/pause, there might be multiple goroutines expanded at the same time, unless you reset the view with the goroutine filters.

To filter as you go, you can use `dlv config` syntax in the DEBUG CONSOLE when stopped at a breakpoint. For example to only display user goroutines and hide anything runtime-related, you can run:
```
dlv config hideSystemGoroutines true
```
or
```
dlv config goroutineFilters -w user
```
More on possible filters at https://github.com/go-delve/delve/blob/master/Documentation/cli/README.md#goroutines
The following filter might be of use when narrowing down to the current goroutine.
```
dlv config goroutineFilters -w running
```
